internal/v2/security.go

```diff
@@ -0,0 +1,119 @@
+package v2
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+)
+
+// SecurityProviderExoscaleV2 represents an Exoscale API V2 security provider.
+type SecurityProviderExoscaleV2 struct {
+	// ReqExpire represents the request expiration duration.
+	ReqExpire time.Duration
+
+	apiKey    string
+	apiSecret string
+}
+
+// NewSecurityProviderExoscaleV2 returns a new Exoscale API V2 security provider to sign API requests using the
+// specified API key/secret.
+func NewSecurityProviderExoscaleV2(apiKey, apiSecret string) (*SecurityProviderExoscaleV2, error) {
+	var provider = SecurityProviderExoscaleV2{
+		ReqExpire: 10 * time.Minute,
+	}
+
+	if provider.apiKey = apiKey; provider.apiKey == "" {
```

Comment:
This syntaxic sugar is hard to process. Is there a way to make it more explicit ?

Reply:
You mean something like

```go
provider.apiKey = apiKey
if provider.apiKey == "" {
	// ...
```

?

Reply:
Something like:
```go
if apiKey == "" {
return nil, errors.New("missing API key")
}

if apiSecret == "" {
return nil, errors.New("missing API key")
}

provided.apiKey = apiKey
provider.apiSecret = apiSecret

return &provider, nil
```

Reply:
Changed in 5aa5488.